airflow/models/mappedoperator.py

```diff
@@ -225,7 +226,7 @@ def expand(self, **mapped_kwargs: "Mappable") -> "MappedOperator":
 class MappedOperator(AbstractOperator):
     """Object representing a mapped operator in a DAG."""
 
-    operator_class: Union[Type["BaseOperator"], str]
+    operator_class: Union[Type["BaseOperator"], Dict[str, Any]]
```

Comment:
This feels like this is doing too many different things now. Maybe we should have two separate attrs instead?

Reply:
The dict variant is actually just an optimisation, we could implement unmap like this instead:

```python
op = SerializedBaseOperator(task_id=self.task_id, _airflow_from_mapped=True)
encoded = SerializedBaseOperator.serialize_operator(self)
data = {k: v for k, v in encoded.items() if k in BaseOperator.get_serialized_fields()}
SerializedBaseOperator.populate_operator(op, data)
return op
```

If we want to avoid re-serialising, I personally prefer using Union here since it’s more obvious that exactly one usage of `operator_class` makes sense for non-serialised and serialised operator. Perhaps we could heavily comment this to make it clearer? Or have a “proper” `SerializedMappedOperator` class instead?

Reply:
Your call.

Reply:
I added a block of text on `operator_class` for this. We can always change this later if it’s too confusing.